src/pybind/mgr/cephadm/module.py

```diff
@@ -1671,11 +1702,20 @@ def remove_service(self, service_name):
                     args.append(
                         ('%s.%s' % (d.daemon_type, d.daemon_id), d.hostname)
                     )
+                    self._remove_key_from_store(d.daemon_type, d.service_name(without_service_type=True))
```

Comment:
this should just be
```
del self.service_spec_store[d.service_name]
```
and drop the _remove_key_from_store() function entirely.

Reply:
I reduced the function to your suggestion (easier for unittesting)